scim2.yaml

```diff
@@ -0,0 +1,848 @@
+    put:
+      tags:
+        - Me Endpoint
+      summary: Return the updated user
+      description: Returns HTTP 404 if the user is not found.
+      operationId: updateUserMe
+      x-wso2-request: |
+        curl -v -k --user [username]:[password] -X PUT -d '{"schemas":[],"name":{"familyName":[last name],"givenName":[name]},"emails":[{"primary":[true/false],"value":[email address],"type":[home/work]},{"value":[email address 2],"type":[home/work]}]}' --header "Content-Type:application/json" https://localhost:9443/scim2/Me
+      x-wso2-curl: |
+        curl -v -k --user kim:kimwso2 -X PUT -d '{"schemas":[],"name":{"familyName":"JacksonJohn","givenName":"Kim"},"userName":"kim","emails":[{"primary":true,"value":"jacksonk@gmail.com","type":"home"},{"value":"jackson_k@wso2.com","type":"work"}],"EnterpriseUser":{"employeeNumber":"123A","manager":{"value":"Taylor"}}}' --header "Content-Type:application/json" https://localhost:9443/scim2/Me
+      x-wso2-response: |
+        {"emails":[{"type":"work","value":"jackson_k@wso2.com"},{"type":"home","value":"jacksonk@gmail.com"}],"meta":{"created":"2018-08-16T17:19:43Z","location":"https://localhost:9443/scim2/Users/f60e6ddd-8d04-411f-92b9-c7ba95fb0fa9","lastModified":"2018-08-16T17:43:17Z","resourceType":"User"},"schemas":["urn:ietf:params:scim:schemas:core:2.0:User","urn:ietf:params:scim:schemas:extension:enterprise:2.0:User"],"EnterpriseUser":{"manager":{"value":"Taylor"},"employeeNumber":"123A"},"roles":[{"type":"default","value":"manager,Internal/everyone,admin"}],"name":{"givenName":"Kim","familyName":"JacksonJohn"},"groups":[{"display":"manager","value":"a43fe003-d90d-43ca-ae38-d2332ecc0f36"}],"id":"f60e6ddd-8d04-411f-92b9-c7ba95fb0fa9","userName":"kim"}
+      consumes:
+        - application/scim+json
+      produces:
+        - application/scim+json
+      parameters:
+        - name: attributes
+          in: query
+          description: SCIM defined attributes parameter.
+          required: false
+          type: string
+        - name: excludedAttributes
+          in: query
+          description: SCIM defined excludedAttribute parameter.
+          required: false
+          type: string
+        - in: body
+          name: body
+          required: false
+          schema:
+            type: string
+      responses:
+        '200':
+          description: User is updated
+        '401':
+          description: Unauthorized
+        '404':
+          description: Valid user is not found
+    delete:
+      tags:
+        - Me Endpoint
+      summary: Delete the authenticated user.
+      description: Returns HTTP 204 if the user is successfully deleted.
+      operationId: deleteUserMe
+      x-wso2-request: |
+         curl -v -k --user  [username]:[password] -X DELETE https://localhost:9443/scim2/Me
+      x-wso2-curl: |
+         curl -v -k --user kim:kimwso2 -X DELETE https://localhost:9443/scim2/Me
+      x-wso2-response: |
+        HTTP/1.1 501 Not Implemented
```

Comment:
Is this correct ?

Reply:
Yes, we didn't implement 'deleteMe' function yet, but the endpoint is there.
https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/blob/a4883981e661ee6122ee50405aef350f1eaeae0c/components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java#L598

https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/blob/a4883981e661ee6122ee50405aef350f1eaeae0c/components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/MeResource.java#L131